src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.Incremental.cs

```diff
@@ -0,0 +1,345 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests
+{
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/52062", TestPlatforms.Browser)]
+    public partial class ConfigurationBindingGeneratorTests : ConfigurationBinderTestsBase
+    {
+        [ActiveIssue("")]
+        [Fact]
+        public async Task RunWithNoDiags_Then_NoEdit()
+        {
+            ConfigBindingGenDriver driver = new ConfigBindingGenDriver();
+
+            ConfigBindingGenResult result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCode);
+            result.AssertHasSourceAndNoDiagnostics();
+
+            result = await driver.RunGeneratorAndUpdateCompilation();
+            result.AssertEmpty();
+        }
+
+        [ActiveIssue("")]
+        [Fact]
+        public async Task RunWithNoDiags_Then_NoOpEdit()
+        {
+            ConfigBindingGenDriver driver = new ConfigBindingGenDriver();
+
+            ConfigBindingGenResult result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCode);
+            result.AssertHasSourceAndNoDiagnostics();
+
+            result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_ReorderedInvocations);
+            result.AssertEmpty();
+
+            result = await driver.RunGeneratorAndUpdateCompilation(BindCallSampleCodeVariant_ReorderedConfigTypeMembers);
+            result.AssertEmpty();
```

Comment:
I'm not sure I understand what these assertions are checking, from the perspective of incremental caching. The two runs can return an empty set of diagnostics, but that doesn't necessarily imply that the intermediate model has been cached. 

You should instead try to query the `GeneratorRunResult` to get an authoritative response from Roslyn as to whether the value was cached or not. Here's an example:

https://github.com/dotnet/runtime/blob/d05415704826f0bd084d2c27f99e4dbe74f62464/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorIncrementalTests.cs#L300-L307